### ISSUE TITLE
Koa Plugin Supresses Application-level Error Events

### ISSUE THREAD

**Issue by mgreystone** [bugsnag/bugsnag-js#1459]
### Describe the bug
The Koa plugin's request handler is suppressing application-level error events. It instead calls `app.onerror` directly, which is Koa's out-of-the-box error handling & not as robust as an error event with `ctx`. This makes it awkward for us to add additional error handling, such as verbose logging.

### Steps to reproduce
1. Create a new koa service following [bugsnag's documentation](https://docs.bugsnag.com/platforms/javascript/koa/)
2. Add an additional error handler

    ```js
    app.on('error', (err, ctx) => console.log('My Custom Error Handler', err))
    ```
3. Throw an error inside a handler

   ```js
    app.use(ctx => { throw new Error('oops, something went wrong') })
    ```

4. Note the custom error handler created in step 2 is never invoked.

### Environment
* Bugsnag version: 7.10.4
* Server framework version (if any):
    * Koa: 7.10.0

### Example Repo 
https://github.com/mgreystone/demo-koa-bugsnag-error-events

### Example code snippet 

```js
const Bugsnag = require('@bugsnag/js')
const BugsnagPluginKoa = require('@bugsnag/plugin-koa')
const Koa = require('koa')

if (!process.env.BUGSNAG_API_KEY) {
  console.error('Missing $BUGSNAG_API_KEY')
  process.exit(1)
}

Bugsnag.start({
  apiKey: process.env.BUGSNAG_API_KEY,
  plugins: [BugsnagPluginKoa]
})

const bugsnagKoaMiddleware = Bugsnag.getPlugin('koa')

const app = new Koa()
app.on('error', customLogger)
app.on('error', bugsnagKoaMiddleware.errorHandler)
app.use(bugsnagKoaMiddleware.requestHandler)
app.use(handler)
app.listen(3001)

function handler () {
  throw new Error('oh no, something went wrong')
}

// THIS IS NEVER INVOKED
function customLogger (err, ctx) {
  console.error(`My Custom Logger: Error with ${ctx.req.method} ${ctx.req.url}`, err)
}

```

**Comment by yousif-bugsnag** [comment]
Hi @mgreystone, thanks for the report and the clear reproduction example! We will look to get this fixed as soon as priorities allow.

**Comment by johnkiely1** [comment]
Hi @mgreystone. Just to let you know this has now been fixed in [v7.11.0](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.11.0) of the notifier.  

Thanks again for raising this.

**Comment by mgreystone** [comment]
@johnkiely1 Thanks. The issue has been resolved on my end in v7.11.0.

I can still get past that & reproduce the original bug if i try to be clever & combine the two error handlers like so:

```js
withBugsnag.on('error', (err, ctx) => {
  customLogger(err, ctx)
  bugsnagKoaMiddleware.errorHandler(err, ctx)
})
```

I can workaround by breaking out into separate handlers, but it still seems unexpected to me. I wanted to mention it.

Thanks again.
